nipype/pipeline/engine/nodes.py

```diff
@@ -1125,6 +1125,8 @@ def _make_nodes(self, cwd=None):
                     fieldvals = filename_to_list(getattr(self.inputs, field))
                 logger.debug('setting input %d %s %s', i, field, fieldvals[i])
                 setattr(node.inputs, field, fieldvals[i])
+            node._interface.num_threads = self._interface.num_threads
```

Comment:
so the issue is with this line:

https://github.com/nipy/nipype/blob/master/nipype/pipeline/engine/nodes.py#L173

therefore in MapNodes, the deepcopy statement should be `n_procs=self._interface.num_threads`

```python
Node(deepcopy(node._interface), 
     name='copynode', 
     n_procs=node._interface.num_threads)._interface.num_threads
```

and also any other attribute that's being set there.